### PR TITLE
autoprune: fix tags returned by autoprune worker (PROJQUAY-8070)

### PR DESCRIPTION
**Pull request by Sunandadadi**
We have the offset being applied before querying for tag_pattern. This is returning wrong results. So, moved offset to be applied on `db_regex_search` if tag_pattern exists.

### CODE REVIEW

File: data/text.py [hunk 1]
```diff
@@ -61,14 +61,19 @@ def match_like(field, search_query):
...
+    return (
+        query.where(field.regexp(pattern)).offset(offset).limit(limit)
+        if matches
+        else query.where(~field.regexp(pattern)).offset(offset).limit(limit)
+    )
 
 
-def regex_sqlite(query, field, pattern, matches=True):
+def regex_sqlite(query, field, pattern, offset, limit, matches=True):
     rows = query.execute()
```
Comment (bcaton85) [inline data/text.py:73]: Can we get a warning and/or a comment saying that the query is going to fetch all rows regardless of what offset and limit is set to?

File: workers/test/test_autopruneworker.py [hunk 2]
```diff
@@ -711,24 +711,29 @@ def test_registry_prune_invalid_policy(initialized_db):
...
         (
             ["match1", "match2", "match3", "match4", "match5"],
             ["match1", "match2", "match3", "match4", "match5"],
             False,
         ),
+        (
+            ["test1", "test2", "test3", "test4", "test5"],
+            ["test1", "test2", "test3"],
+            False,
+        ),
```
Comment (bcaton85) [inline workers/test/test_autopruneworker.py:736]: I have some additional test cases that i'll send over

File: data/text.py (hunk 1 above)
Comment (Sunandadadi) [reply data/text.py:74]: good call, added.

**Review by bcaton85 — APPROVED** [review]

**Pull request merged**